### PR TITLE
types: start *Events tests

### PR DESCRIPTION
**Pull request by odeke-em**
Updates https://github.com/tendermint/tendermint/issues/693

Removed the seemingly now defunct EventData*.Assert* methods

Make a sequence logger to fire events then check their
outputs ensuring that they fire in sequence and pump
out data with the respective wanted event name strings.

### CODE REVIEW

File: types/events.go [hunk 1]
```diff
@@ -140,14 +140,6 @@ type EventDataVote struct {
 	Vote *Vote
 }
 
-func (_ EventDataNewBlock) AssertIsTMEventData()       {}
```
Comment (ebuchman) [inline types/events.go]: nice

**Comment by ebuchman** [comment]
I applaud the effort here but the test feels to fragile. Can we reduce the dependence on hardcoded strings ?

**Comment by odeke-em** [comment]
@ebuchman could you please help with a suggestion to reduce the dependency on strings? Or how would you test FireEvent?

**Comment by ebuchman** [comment]
Can we not use `EventStringXxx()` and json marshalling?

**Comment by odeke-em** [comment]
`Can we not use EventStringXxx() and json marshalling?`
@ebuchman that's exactly what I did for the test https://github.com/tendermint/tendermint/pull/749/files#diff-32c6067f84ad3192c426e89a0eb16cfcR23
screen shot 2017-10-18 at 11 31 15 pm

and those strings are the expected outputs

**Comment by ebuchman** [comment]
Yeh I see, we're mocking the fireable with that. It just feels a bit excessive and difficult to maintain. At the least the strings should be paired with the structs in one list. But why not use the tmlibs/events.EventSwitch and check the outputs from there against the json.Marshals so we don't have to write strings manually?

**Comment by odeke-em** [comment]
@ebuchman thanks for the feedback. Alright, I see, I'll make another update but if I haven't simplified it more, could you please help me take over it? This way we can spend less back and forth on reviews and comments and we can have more throughput in coverage.

**Comment by ebuchman** [comment]
Sure thing - have just been out of pocket this week from travelling. Back on it later today.

Thanks.

**Comment by melekes** [comment]
Once #568 is merged, these tests probably will lose their value

**Comment by ebuchman** [comment]
Oh of course. Let's reconsider them after then.

**Pull request closed without merging**